src/pages/weather-dashboard/components/weather-chart.tsx

```diff
@@ -0,0 +1,232 @@
+    hourlyWeather.time.slice(0, hoursToShow).map((time, index) => ({
+      x: formatChartTime(time, timeRange),
+      y: hourlyWeather.precipitation?.[index] || 0,
+    }));
+
+  // Prepare precipitation probability data for the chart
+  const precipitationProbData =
+    hourlyWeather.precipitation_probability &&
+    hourlyWeather.time.slice(0, hoursToShow).map((time, index) => ({
+      x: formatChartTime(time, timeRange),
+      y: hourlyWeather.precipitation_probability?.[index] || 0,
+    }));
+
+  const tempUnit = temperatureUnit === 'celsius' ? '°C' : '°F';
+  const precipUnit = precipitationUnit === 'mm' ? 'mm' : 'in';
+
+  return (
+    <Container
+      header={
+        <Header
+          variant="h2"
+          actions={
+            <SpaceBetween direction="horizontal" size="xs">
+              <SegmentedControl
+                selectedId={timeRange}
+                onChange={({ detail }) => setTimeRange(detail.selectedId as '24h' | '48h' | '7d')}
+                options={[
+                  { id: '24h', text: '24 hours' },
+                  { id: '48h', text: '48 hours' },
+                  { id: '7d', text: '7 days' },
+                ]}
+              />
+            </SpaceBetween>
+          }
+        >
+          Weather Charts
+        </Header>
+      }
+    >
+      <SpaceBetween size="l">
+        <ColumnLayout columns={1}>
+          <LineChart
+            series={[
+              {
+                title: `Temperature (${tempUnit})`,
+                type: 'line',
+                data: temperatureData,
+                valueFormatter: value => `${value}${tempUnit}`,
+              },
+            ]}
+            xScaleType="categorical"
+            yDomain={[
+              Math.min(...hourlyWeather.temperature_2m.slice(0, hoursToShow)) - 5,
+              Math.max(...hourlyWeather.temperature_2m.slice(0, hoursToShow)) + 5,
+            ]}
+            i18nStrings={{
+              filterLabel: 'Filter displayed data',
+              filterPlaceholder: 'Filter data',
+              filterSelectedAriaLabel: 'selected',
+              legendAriaLabel: 'Legend',
+              chartAriaRoleDescription: 'line chart',
+              xAxisAriaRoleDescription: 'x axis',
+              yAxisAriaRoleDescription: 'y axis',
+              yTickFormatter: value => `${value}${tempUnit}`,
+            }}
+            ariaLabel="Temperature chart"
+            height={300}
+            hideFilter
+            hideLegend
+            xTitle="Time"
+            yTitle={`Temperature (${tempUnit})`}
+            empty={
+              <Box textAlign="center" color="inherit">
+                <b>No data available</b>
+                <Box variant="p" color="inherit">
+                  There is no temperature data available for the selected time range.
+                </Box>
+              </Box>
+            }
+          />
+        </ColumnLayout>
+
+        {precipitationData && precipitationProbData && (
```

Comment:
@builderio-bot move this to be in its own component in a new file and use that compnent here